scipy/sparse/coo.py

```diff
@@ -278,31 +275,26 @@ def tocsc(self, copy=False):
             return csc_matrix(self.shape, dtype=self.dtype)
         else:
             M,N = self.shape
+            self.sum_duplicates()
             idx_dtype = get_index_dtype((self.col, self.row),
                                         maxval=max(self.nnz, M))
+            row = self.row.astype(idx_dtype, copy=False)
+            col = self.col.astype(idx_dtype, copy=False)
```

Comment:
No, sparsetools is actually working ok.

The failing tests are a bit violent as they monkeypatch get_index_dtype with:
```
        def new_get_index_dtype(arrays=(), maxval=None, check_contents=False):
            return (np.int32, np.int64)[counter.randint(2)]
```
which sometimes downcasts. iirc, this is intentional, but maybe it should be made to follow same downcast semantics as the actual get_index_dtype function... Anyway, that's for later.